base/compiler/abstractinterpretation.jl

```diff
@@ -42,12 +42,24 @@ function abstract_call_gf_by_type(@nospecialize(f), argtypes::Vector{Any}, @nosp
         splitsigs = switchtupleunion(atype)
         applicable = Any[]
         for sig_n in splitsigs
-            xapplicable = _methods_by_ftype(sig_n, max_methods, sv.params.world, min_valid, max_valid)
+            (xapplicable, min_valid[1], max_valid[1]) =
+                get(sv.matching_methods_cache, sig_n) do
```

Comment:
This can use `get!`; then the explicit assignment into the cache below is not needed.

Reply:
Oh, thanks--amended.